### PR TITLE
Implement WithMaxSeconds for GlpkSolver

### PR DESCRIPTION
**Pull request by mfuhr**
Implement `WithMaxSeconds` for `GlpkSolver` and recognize when the solution is suboptimal.

I used `CbcSolver` as a template but I noticed that its `command_name` and `with_temp_solution_file` methods set `seconds` to `None` instead of copying it from `&self`.  I don't know if that's intentional but I wasn't expecting it, so for `GlpkSolver` I have those methods copy `seconds`.

### CODE REVIEW

File: src/solvers/glpk.rs [hunk 1]
```diff
@@ -118,18 +123,40 @@ impl SolverWithSolutionParsing for GlpkSolver {
...
             solution_file.into(),
-        ]
+        ];
+
+        for (name, value) in [("--tmlim", self.max_seconds())].iter() {
+            if let Some(val) = value {
+                args.push(name.into());
+                args.push(val.to_string().into());
+            }
+        }
```
Comment (lovasoa) [inline src/solvers/glpk.rs:157]: Isn't that more complicated than it needs to be ?

Could we have 

```suggestion
        if let Some(seconds) = self.max_seconds() {
            args.push("--tmlim".into());
            args.push(seconds.to_string().into());
        }
```

Comment (mfuhr) [reply src/solvers/glpk.rs:157]: I copied that approach from `CbcSolver`, although that code actually uses both approaches and ends up setting the timeout arguments twice.  I figured the iterator would simplify the code as more options are added (e.g., the MIP gap option that  just proposed) but I'd be happy to change it.

Comment (lovasoa) [reply src/solvers/glpk.rs:157]: ok, I'm fine with that.

**Comment by lovasoa** [comment]
published as v0.0.5

**Pull request merged**